src/darwin/Framework/CHIP/MTRDeviceControllerDelegateBridge.h

```diff
@@ -39,6 +40,7 @@ class MTRDeviceControllerDelegateBridge : public chip::Controller::DevicePairing
     void OnCommissioningComplete(chip::NodeId deviceId, CHIP_ERROR error) override;
 
 private:
+    MTRDeviceController * _Nullable mController;
```

Comment:
Does this need to be `__weak` ? I think the `MTRDeviceController` holds a strong reference to this bridge?

Reply:
This bridge is not reference counted; it's a C++ object, not an Objective-C object.

So yes, the controller holds an owning reference (allocated with `new`) to the bridge, and deletes the bridge in `cleanup`.  As a result, in practice it doesn't matter that much whether this is a weak ref or not, but you're right that it's probably more idiomatic to make it weak.

Reply:
Done.